### PR TITLE
Undo and redo feature for Update command

### PR DESCRIPTION
**Pull request by lightz96**

### CODE REVIEW

**Comment by lightz96** [comment]
@lye-jw I shifted updateFilteredTransactionList() function caller in UpdateCommand#execute to ModelManager#setTransaction

File: src/test/java/thrift/logic/commands/UpdateCommandTest.java [hunk 1]
```diff
@@ -140,6 +143,62 @@ public void execute_invalidTransactionIndexFilteredList_failure() {
...
+        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(),
+                new PastUndoableCommands());
+        expectedModel.setTransaction(lastTransaction, updatedTransaction);
+        assertCommandSuccess(updateCommand, model, expectedMessageUpdated + expectedMessageOriginal, expectedModel);
+
+        //test undo
+        expectedModel.setTransaction(updatedTransaction, lastTransaction);
+        assertUndoCommandSuccess(updateCommand, model, expectedModel);
+
+        //tesst redo
```
Comment (lye-jw) [inline src/test/java/thrift/logic/commands/UpdateCommandTest.java]: Extremely tiny comment typo.

**Review by lye-jw — APPROVED** [review]
LGTM!

File: src/test/java/thrift/logic/commands/UpdateCommandTest.java (hunk 1 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/UpdateCommandTest.java]: Resolves 9831c05

**Pull request merged**